### PR TITLE
add config for cluster-api-provider-libvirt repo

### PR DESCRIPTION
**Pull request by paulfantom**
This adds new repository to CI system. PR is based on work done previously for cluster-api-provider-aws.

In current state we do not expect to build an image yet so there is no postsubmit stage.

/CC: @petr-muller @enxebre

### CODE REVIEW

File: ci-operator/jobs/openshift/cluster-api-provider-libvirt/openshift-cluster-api-provider-libvirt-presubmits-release-3.11.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+presubmits:
+  openshift/cluster-api-provider-libvirt:
+  - agent: kubernetes
+    always_run: true
+    branches:
+    - master
```
Comment (petr-muller) [inline ci-operator/jobs/openshift/cluster-api-provider-libvirt/openshift-cluster-api-provider-libvirt-presubmits-release-3.11.yaml]: Should this be `release-3.11`?

File: ci-operator/jobs/openshift/cluster-api-provider-libvirt/openshift-cluster-api-provider-libvirt-presubmits-release-3.11.yaml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    context: ci/prow/unit
+    decorate: true
+    name: pull-ci-origin-cluster-api-provider-libvirt-unit-release-3.11
```
Comment (petr-muller) [inline ci-operator/jobs/openshift/cluster-api-provider-libvirt/openshift-cluster-api-provider-libvirt-presubmits-release-3.11.yaml]: For the sake of consistency, we suggest the `pull-ci-$org-$repo-$branch-$test` template for names (not `...-$test-$branch`)

Comment (paulfantom) [reply ci-operator/jobs/openshift/cluster-api-provider-libvirt/openshift-cluster-api-provider-libvirt-presubmits-release-3.11.yaml]: This name is consistent with cluster-api-provider-aws: https://github.com/openshift/release/blob/master/ci-operator/jobs/openshift/cluster-api-provider-aws/openshift-cluster-api-provider-aws-presubmits-release-3.11.yaml#L9 on which I based this PR.

File: ci-operator/jobs/openshift/cluster-api-provider-libvirt/openshift-cluster-api-provider-libvirt-presubmits-release-3.11.yaml (hunk 1 above)
Comment (paulfantom) [reply ci-operator/jobs/openshift/cluster-api-provider-libvirt/openshift-cluster-api-provider-libvirt-presubmits-release-3.11.yaml]: Fixed :+1:

File: ci-operator/jobs/openshift/cluster-api-provider-libvirt/openshift-cluster-api-provider-libvirt-presubmits-release-3.11.yaml (hunk 2 above)
Comment (petr-muller) [reply ci-operator/jobs/openshift/cluster-api-provider-libvirt/openshift-cluster-api-provider-libvirt-presubmits-release-3.11.yaml]: \shrug

We'll standardize the names soon with the generator anyway ;)

**Pull request merged**